src/Tokenizer.elm

```diff
@@ -1,93 +1,151 @@
-module Tokenizer exposing (..)
-
+module Tokenizer exposing (fromMaybeInt, isBoolean, operator, operators, stringToBoolean, stringToInt, tokenize, tokenizeLine, tokenizePrint)
 
 import Debug exposing (toString)
 import Types exposing (..)
 
+
+
 -------------------------------------- Tokenizer --------------------------------------
 
+
 operator : String -> Token
 operator str =
-            case str of
-                "" -> TokInvalid
-                other ->
-                    if (str == "+")
-                         then TTSC (TTSCInt TokPlus)
-                     else if (str == "-")
-                         then TTSC (TTSCInt TokMinus)
-                     else if (str == "*")
-                         then TTSC (TTSCInt TokTimes)
-                     else if (str == "=")
-                         then TokAssign
-                     else if (str == "==")
-                          then TTSC (TTSCBool TokEq)
-                     else if (str == "&&")
-                          then TTSC (TTSCBool TokAnd)
-                     else if (str == "||")
-                          then TTSC (TTSCBool TokOr)
-                     else if (str == "?")
-                         then TokHole
-                     else TokInvalid
-
-operators = ["+", "-", "*", "=", "==", "&&", "||", "?"]
+    case str of
+        "" ->
+            TokInvalid
+
+        other ->
+            if str == "+" then
+                TTSC (TTSCInt TokPlus)
+
+            else if str == "-" then
+                TTSC (TTSCInt TokMinus)
+
+            else if str == "*" then
+                TTSC (TTSCInt TokTimes)
+
+            else if str == "=" then
+                TokAssign
+
+            else if str == "==" then
+                TTSC (TTSCBool TokEq)
+
+            else if str == "&&" then
+                TTSC (TTSCBool TokAnd)
+
+            else if str == "||" then
+                TTSC (TTSCBool TokOr)
+
+            else if str == "?" then
+                TokHole
+
+            else
+                TokInvalid
+
+
+operators =
+    [ "+", "-", "*", "=", "==", "&&", "||", "?" ]
+
 
 isBoolean : String -> Bool
 isBoolean x =
-  case x of
-      "True" -> True
-      "False" -> True
-      other -> False
+    case x of
+        "True" ->
+            True
+
+        "False" ->
+            True
+
+        other ->
+            False
+
 
 stringToBoolean : String -> Bool
 stringToBoolean x =
-  case x of
-      "True"  -> True
-      "False" -> False
-      _       -> False                                    -- should never get called
+    case x of
+        "True" ->
+            True
+
+        "False" ->
+            False
+
+        _ ->
+            False
+
+
+
+-- should never get called
+
 
 stringToInt : String -> Int
-stringToInt x = fromMaybeInt(String.toInt x)
+stringToInt x =
+    fromMaybeInt (String.toInt x)
+
 
```

Comment:
We can do away with this function if we use elm's Maybe.withDefault : https://package.elm-lang.org/packages/elm/core/latest/Maybe#withDefault

Not necessary, but might lead to more readable code